uaclient/tests/test_gpg.py

```diff
@@ -24,74 +24,39 @@ def home_dir(tmpdir):
-        src_keyring.write("")
+        """Raise UserFacingError when key_id is not in source_keyring_dir."""
+        src_keyring_dir = tmpdir.strpath
         destination_keyfile = tmpdir.join("destination_keyfile").strpath
-        # known valid gpg key which will not exist in src_keyring
+        # known valid gpg key which will not exist in source_keyring_dir
         key_id = "7CF05F218D72BB5B945693CA6FE15DD1E3510761"
         with pytest.raises(exceptions.UserFacingError) as excinfo:
-            gpg.export_gpg_key_from_keyring(
+            gpg.export_gpg_key_from_keyring_dir(
                 key_id=key_id,
-                source_keyring_file=src_keyring.strpath,
+                source_keyring_dir=src_keyring_dir,
                 destination_keyfile=destination_keyfile,
             )
 
         error_msg = "GPG key '{}' not found in {}".format(
-            key_id, src_keyring.strpath
+            key_id, src_keyring_dir
         )
         assert error_msg in str(excinfo.value)
         assert not os.path.exists(destination_keyfile)
 
-    def test_user_facing_error_on_invalid_keyring_packet(
-        self, caplog_text, home_dir, tmpdir
-    ):
-        """Raise UserFacingError on invalid keyring packet. Log cmd error."""
-        source_keyring = tmpdir.join("keyring").strpath
-        destination_keyfile = tmpdir.join("destination_key").strpath
-        key_id = "3CB3DF682220A643B43065E9B30EDAA63D8F61D0"
-        # Create bogus keyring with invalid packet which causes gpg errors
-        tmpdir.join("keyring").write(b"\n", mode="wb")
-        with pytest.raises(exceptions.UserFacingError) as excinfo:
-            gpg.export_gpg_key_from_keyring(
-                key_id=key_id,
-                source_keyring_file=source_keyring,
-                destination_keyfile=destination_keyfile,
-            )
-        msg = "Failed running command 'gpg --output {}".format(
-            destination_keyfile
-        )
-        assert "ERROR    {}".format(msg) in caplog_text()
-        assert not os.path.exists(destination_keyfile)
-        msg = "Unable to export GPG keys from keyring {}".format(
-            source_keyring
-        )
-        assert msg == str(excinfo.value)
-
-    def test_export_single_key_from_keyring(self, home_dir, tmpdir):
+    def test_export_single_key_from_keyring_dir(self, home_dir, tmpdir):
         """Only a single key is exported from a multi-key source keyring."""
-        source_keyring = tmpdir.join("keyring").strpath
+        source_key1 = tmpdir.join("ua-{}.gpg".format(data.GPG_KEY1_ID)).strpath
+        source_key2 = tmpdir.join("ua-{}.gpg".format(data.GPG_KEY2_ID)).strpath
         destination_keyfile = tmpdir.join("destination_key").strpath
         # Create keyring with both ESM and CC-EAL2 keys
-        with open(source_keyring, "wb") as stream:
-            stream.write(data.GPG_KEY1 + data.GPG_KEY2)
-        gpg.export_gpg_key_from_keyring(
+        with open(source_key1, "wb") as stream:
```

Comment:
A nit, but if you don't call `strpath` on the `tmpdir.join` return, I think you could do `source_key1.write(data.GPG_KEY1, "wb")` and save a bit of vertical space.

Reply:
excellent suggestion.